### PR TITLE
feat(identities): add metadata editing capabilities

### LINKED ISSUE #4458: Identity Edit Metadata
Add metadata editing capability to identities, accessible from the context menu.

## **Acceptance Criteria:**

*  New menu item in context menu: "Edit metadata..."
*  Dialog opens with metadata-setup component
*  Full replacement model with warning
*  Validation (JSON syntax, 1MB size limit)
*  Success/error notifications
*  Optimistic UI updates

### PR DESCRIPTION
**Pull request by mcstepp**
## What does this PR do?

Added Identity Metadata Editing
**New Functionality:**

- Added "Edit metadata..." option to identity context menu with divider separator
- Created edit metadata dialog matching key edit dialog styling with:
- Identity card showing fingerprint icon, ID, and external ID (truncated with tooltip)
- Toggle switch to enable/disable metadata
- Full JSON validation and 1MB size limit
- Success/error notifications
- Updated create identity dialog to use shared metadata component with toggle switch
- Added TRPC mutation endpoint identity.update.metadata with validation and audit logging
- Fixed action column click preventing unwanted row navigation

**Code Refactoring:**
Extracted metadata components to shared location components/dashboard/metadata/:

- metadata-setup.tsx - Reusable component with required entityType prop
- protection-switch.tsx - Shared toggle switch
- Moved metadata schema to lib/schemas/metadata.ts
- Updated all existing usages (key creation, key editing) to use shared components
- Deleted duplicate metadata-setup component from create-key directory

Screenshot 2025-12-11 at 1 41 40 PM
Screenshot 2025-12-11 at 1 41 51 PM
Screenshot 2025-12-11 at 1 42 04 PM

Fixes #4458 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

1. Create Identity with Metadata

- Click "Create Identity" button
- Enter an external ID (e.g., user@example.com)
- Toggle "Metadata" switch ON
- Enter valid JSON (or click Format button)
- Click "Create Identity"
- Verify identity appears in list

2. Edit Metadata - Add to Identity Without Metadata

- Create an identity without metadata
- Click three-dot menu (⋮) → "Edit metadata..."
- Verify dialog shows identity card with fingerprint icon, ID, and external ID
- Verify metadata toggle is OFF
- Toggle metadata ON
- Enter JSON: {"plan": "pro", "userId": "123"}
- Click "Update metadata"
- Verify success toast appears
- Verify identity list refreshes

3. Edit Metadata - Update Existing Metadata

- Create an identity with metadata
- Click three-dot menu → "Edit metadata..."
- Verify metadata toggle is ON
- Verify existing metadata is pre-populated in textarea
- Modify the JSON
- Click "Update metadata"
- Verify changes are saved

4. Edit Metadata - Remove Metadata

- Find an identity with metadata
- Click three-dot menu → "Edit metadata..."
- Toggle metadata OFF
- Click "Update metadata"
- Verify metadata is removed

5. Validation Tests

- Try to save invalid JSON → Should show error
- Try to save empty textarea with metadata enabled → Should show validation error
- Click Format button with valid JSON → Should prettify JSON
- Toggle metadata OFF/ON → Textarea should disable/enable

6. UI/UX Tests

- Long external ID (like long email) → Should truncate with tooltip
- Click outside dialog → Should close
- Click action menu → Should NOT navigate to detail page

7. Regression Test
- All Creating key with metadata/editing key with metadata functionality should still work

**Edge Cases**

- Identity with very long ID/external ID
- Metadata at ~1MB size limit
- Special characters in JSON
- Opening/closing dialog multiple times

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [ ] Ran `make fmt` on `/go` directory
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

File: apps/dashboard/app/(app)/[workspaceSlug]/apis/[apiId]/_components/create-key/components/protection-switch.tsx [hunk 1]
```diff
@@ -1,50 +1,2 @@
...
-          {...switchProps}
-        />
-      </div>
-    );
-  },
-);
-
-ProtectionSwitch.displayName = "ProtectionSwitch";
+// Re-export from shared location for backwards compatibility
+export { ProtectionSwitch } from "@/components/dashboard/metadata/protection-switch";
```
Comment (mcstepp) [inline apps/dashboard/app/(app)/[workspaceSlug]/apis/[apiId]/_components/create-key/components/protection-switch.tsx]: Im removing this

File: apps/dashboard/lib/schemas/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  data: z
+    .string({
+      required_error: "Metadata is required",
+      invalid_type_error: "Metadata must be a JSON",
+    })
+    .trim()
+    .min(2, { message: "Metadata must contain valid JSON" })
+    .max(65534, {
+      message: "Metadata cannot exceed 65535 characters (text field limit)",
+    })
```
[the remark this answers is not in the document]
Comment (mcstepp) [reply apps/dashboard/lib/schemas/metadata.ts:36]: Leave this for now. We're going to revisit the requirement

File: apps/dashboard/lib/trpc/routers/identity/updateMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+      // Check 1MB size limit (1048576 bytes)
+      const metadataString = JSON.stringify(meta);
+      const sizeInBytes = new TextEncoder().encode(metadataString).length;
+      if (sizeInBytes > 1048576) {
+        throw new TRPCError({
+          message: `Metadata size (${Math.round(sizeInBytes / 1024)}KB) exceeds the 1MB limit.`,
+          code: "BAD_REQUEST",
+        });
+      }
```
[the remark this answers is not in the document]
Comment (mcstepp) [reply apps/dashboard/lib/trpc/routers/identity/updateMetadata.ts:37]: maybe later

**Review by perkinsjr — APPROVED** [review]
LGTM
